pkg/clicmd/prototype_use.go

```diff
@@ -84,34 +87,80 @@ ks prototype use single-port-deployment nginx-depl \
 `
 )
 
-func newGenerateCmd(a app.App) *cobra.Command {
-	cmd := newPrototypeUseCmd(a)
+func newGenerateCmd(fs afero.Fs) *cobra.Command {
+	cmd := newPrototypeUseCmd(fs)
 	cmd.Use = "generate <prototype-name> <component-name> [type] [parameter-flags]"
 
 	return cmd
 }
 
-func newPrototypeUseCmd(a app.App) *cobra.Command {
+func newPrototypeUseCmd(fs afero.Fs) *cobra.Command {
 	prototypeUseCmd := &cobra.Command{
-		Use:                "use <prototype-name> <componentName> [type] [parameter-flags]",
-		Short:              protoShortDesc["use"],
-		Long:               prototypeUseLong,
-		Example:            prototypeUseExample,
+		Use:     "use <prototype-name> <componentName> [type] [parameter-flags]",
+		Short:   protoShortDesc["use"],
+		Long:    prototypeUseLong,
+		Example: prototypeUseExample,
+		FParseErrWhitelist: cobra.FParseErrWhitelist{
+			UnknownFlags: true,
+		},
 		DisableFlagParsing: true,
-		RunE: func(cmd *cobra.Command, rawArgs []string) error {
-			if len(rawArgs) == 1 && (rawArgs[0] == "--help" || rawArgs[0] == "-h") {
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.DisableFlagParsing = false
+			err := cmd.ParseFlags(args)
+
+			if err != nil {
+				return cmd.FlagErrorFunc()(cmd, err)
+			}
+
+			if cmd.Flags().Lookup("help").Changed {
 				return cmd.Help()
 			}
 
+			args = removePersistentFlags(cmd.Flags(), args)
+
 			m := map[string]interface{}{
-				actions.OptionApp:       a,
-				actions.OptionArguments: rawArgs,
-				// We don't pass flagTLSSkipVerify because flag parsing is disabled
+				actions.OptionFs:        fs,
+				actions.OptionArguments: args,
 			}
+			addGlobalOptions(m)
 
 			return runAction(actionPrototypeUse, m)
 		},
 	}
 
 	return prototypeUseCmd
 }
+
+func removePersistentFlags(flags *pflag.FlagSet, args []string) []string {
+	flags.Visit(func(flag *pflag.Flag) {
+		for i, arg := range args {
+			if len(arg) == 0 || arg[0] != '-' || len(arg) == 1 {
```

Comment:
`len(arg) < 2`?

Reply:
heh - yeah - just borrowed some of this code from cobra itself because i wanted to make sure i'm parsing flags the same way they do. i'll tweak it.

Reply:
actually it needs to be this way but i'll add a comment